sql/catalyst/src/main/scala/org/apache/spark/sql/internal/connector/PredicateUtils.scala

```diff
@@ -132,4 +133,18 @@ private[sql] object PredicateUtils {
       case _ => None
     }
   }
+
+  def toV1(
+      predicates: Array[Predicate],
+      skipIfNotConvertible: Boolean): Array[Filter] = {
+    predicates.flatMap { predicate =>
+      val filter = toV1(predicate)
+      if (filter.isEmpty) {
+        if (!skipIfNotConvertible) {
+          throw QueryCompilationErrors.unsupportedPredicateToFilterConversionError(predicate.name())
+        }
+      }
+      filter
+    }.toArray
```

Comment:
This `.toArray` seems a little redundant, or is it for Scala 2.13 compatibility?

Reply:
I thought Scala 2.13 needs this `toArray` but it turned out not to be the case. I have it removed.

Reply:
Yes, I double checked that Scala 2.13 doesn't need this `toArray`.  Moreover, the redundant 'toArray' will make the performance worse when using Scala 2.13 due `ArrayOps.toArray` will always do a memory copy even if the item Type has not changed , but Scala 2.12 will directly return the original array when the item Type is unchanged.